### PR TITLE
Migrate to GH Actions + `elm-tooling` and expose `subsequencesNonEmpty` 🆕

### LINKED ISSUE #153: Tail call optimiziations
@jfmengels did some tail call optimizing on a few functions a wall back (Thanks @jfmengels ). He noted that with this command:
```
npx elm-review --template jfmengels/elm-review-performance/example --rules NoUnoptimizedRecursion
```
many more could be fixed. I ran it myself a while back and found ~10 functions that could be optimized.

It would be cool to go through those. Fixing them one at a time and measuring the performance gain.

### PR DESCRIPTION
**Pull request by kutyel**
This is just a general migration to GH Actions and the latest version of `elm-test` to make it easier to maintain the library in the future. Also, CI will now be run on PRs! (which is desirable for contributions 😉 ).

This setup is identical to the one for [`maybe-extra`](https://github.com/elm-community/maybe-extra/blob/master/package.json) and so I think unifying this for some `elm-community` packages could be beneficial 💡 

Also this uses the latest version of `elm-test-rs@3.0.0` 🚀 

Please feel free to add the `hacktoberfest-accepted` label to this PR 😉

### CODE REVIEW

**Comment by kutyel** [comment]
To verify that the GH action succeeded, you can check my fork https://github.com/kutyel/list-extra/actions/runs/3320767331/jobs/5487533854

File: src/List/Extra.elm [hunk 1]
```diff
@@ -1,9 +1,9 @@
 module List.Extra exposing
     ( last, init, getAt, uncons, unconsLast, maximumBy, maximumWith, minimumBy, minimumWith, andMap, andThen, reverseMap, takeWhile, dropWhile, unique, uniqueBy, allDifferent, allDifferentBy, setIf, setAt, remove, updateIf, updateAt, updateIfIndex, removeAt, removeIfIndex, filterNot, swapAt, stableSortWith
     , intercalate, transpose, subsequences, permutations, interweave, cartesianProduct, uniquePairs
-    , Step(..), foldl1, foldr1, indexedFoldl, indexedFoldr, stoppableFoldl
+    , foldl1, foldr1, indexedFoldl, indexedFoldr, Step(..), stoppableFoldl
     , scanl, scanl1, scanr, scanr1, mapAccuml, mapAccumr, unfoldr, iterate, initialize, cycle, reverseRange
-    , splitAt, splitWhen, takeWhileRight, dropWhileRight, span, break, stripPrefix, group, groupWhile, inits, tails, select, selectSplit, gatherEquals, gatherEqualsBy, gatherWith
+    , splitAt, splitWhen, takeWhileRight, dropWhileRight, span, break, stripPrefix, group, groupWhile, inits, tails, select, selectSplit, gatherEquals, gatherEqualsBy, gatherWith, subsequencesNonEmpty
```
Comment (Janiczek) [inline src/List/Extra.elm:6]: You probably forgot to mention a new function. Or should it be in a separate PR?

Comment (kutyel) [reply src/List/Extra.elm:6]: Right, added it to the title of the PR thanks! looked like an abandoned function but might be useful for someone else! 🙏🏻

**Comment by Janiczek** [comment]
(I see your "requested a review" but I'm not a maintainer :) @Chadtech is)

**Comment by kutyel** [comment]
@Janiczek @Chadtech also open to discuss if it is valuable to add the `package-lock.json` file or not, in this case it would be useful for caching (in the GH action 😉 )

**Comment by Chadtech** [comment]
Thank you very much! This was much needed! ❤️
